guides/plugins/plugins/framework/data-handling/using-database-events.md

```diff
@@ -12,7 +12,137 @@ This guide is built upon the [Plugin base guide](../../plugin-base-guide.md), bu
 
 Furthermore you should have a look at our [Listening to events](../../plugin-fundamentals/listening-to-events.md) guide since we are subscribing to events in this guide.
 
-## Event overview
+## General event overview
+
+The events below are dispatched during certain DAL operations, they are not necessarily associated with a particular entity, rather they are triggered with batches of commands.
+
+| Event                                                                   | Description                                                                                        |
+|:------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------|
+| `Shopware\Core\Framework\DataAbstractionLayer\Event\EntityWriteEvent`   | Before a batch of commands has been written to storage. Written means inserted, updated or deleted |
+| `Shopware\Core\Framework\DataAbstractionLayer\Event\EntityDeleteEvent`  | Before a batch of delete commands has been executed                                                |
+
+### `Shopware\Core\Framework\DataAbstractionLayer\Event\EntityWriteEvent`
+
+This event allows you to hook in to the process of writing an entity. This includes, creating, updating and deleting entities. You have the possibility to execute code before and after the entity is written via the success and error callbacks. You can call the `addSuccess` or `addError` methods with any PHP callable.
+
+You can use this event to capture state and perform actions and sync data after an entity is written. It could be used for example, to synchronize images to a CDN when they are written, updated or deleted. This event is useful when you need the before state of the entity. For example, the old filename.
+
+Below is an example subscriber listening to the generic entity write event, and logging the ID's of the written entities.
```

Comment:
```suggestion
This event allows you to hook into the process of writing an entity. This includes creating, updating, and deleting entities. You have the possibility to execute the code before and after the entity is written via the "success" and "error" callbacks. You can call the `addSuccess` or `addError` methods with any PHP callable.

You can use this event to capture state, perform actions, and sync data after an entity is written. It could be used, for example, to synchronize images to a CDN when they are written, updated, or deleted. This event is useful when you need the before state of the entity. For example, the old filename.

Below is an example subscriber listening to the generic entity write event and logging the ID's of the written entities.
```